src/core/manager.c

```diff
@@ -3216,7 +3217,8 @@ usec_t manager_get_watchdog(Manager *m, WatchdogType t) {
         if (MANAGER_IS_USER(m))
                 return USEC_INFINITY;
 
-        if (timestamp_is_set(m->watchdog_overridden[t]))
+        if (timestamp_is_set(m->watchdog_overridden[t]) ||
+            (t == WATCHDOG_PRETIMEOUT && m->watchdog_overridden[t] != USEC_INFINITY))
```

Comment:
i don#t grok this check. timestamp_is_set() already returns true if the timestamp is > 0 and < USEC_INFINITY. Thus with this change you only add that we also return the overriden timestamp in case it is set to 0, but if that's the intention here, then it would be nicer to write that explicitly, no?

Reply:
I'm not sure I understand why it needs to be treated differently either. I'll remove this change for now. @curtistklein any comment?